### PR TITLE
Add Graph SLAM documentation and SE(2) example

### PR DESCRIPTION
**Pull request by JeffLIrion**
This is a work in progress.

The Jupyter notebook doesn't render correctly on GitHub, so I moved the write-up into a LaTeX file and included both the .tex source and the generated PDF.  I'll probably remove it altogether from the Jupyter notebook.

TODO: Add code and example problem

Related issue: https://github.com/AtsushiSakai/PythonRobotics/issues/296

### CODE REVIEW

**Comment by AtsushiSakai** [comment]
@JeffLIrion Your doc looks great!!. Please keep working it : ).

**Comment by JeffLIrion** [comment]
### Current status

- [x] Add a few citations to the write-up
- [x] Add code for loading, optimizing, and plotting `SE(2)` datasets
- [ ] Add an example problem with figures
- [ ] Clean up the Jupyter notebook

Any other tasks that you'd like to see completed?

**Comment by AtsushiSakai** [comment]
Thank you for your hard work!!. That is OK. Please let me know when you are ready to review.

**Comment by JeffLIrion** [comment]
This is ready to start the review process!

File: SLAM/GraphBasedSLAM/graphslam/edge/edge_odometry.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2020 Jeff Irion and contributors
+#
+# This file originated from the `graphslam` package:
+#
+#   https://github.com/JeffLIrion/python-graphslam
+
+r"""A class for odometry edges.
```
Comment (AtsushiSakai) [inline SLAM/GraphBasedSLAM/graphslam/edge/edge_odometry.py]: Should it be raw string?

Comment (AtsushiSakai) [reply SLAM/GraphBasedSLAM/graphslam/edge/edge_odometry.py]: You are using raw string in several docstrings. Is it for latex math?

Comment (JeffLIrion) [reply SLAM/GraphBasedSLAM/graphslam/edge/edge_odometry.py]: Yes, it was for LaTeX math, although as you found it isn't always needed.  I could change these to regular strings if you'd like.

**Comment by JeffLIrion** [comment]
1. This repo already have  graph_based_slam.py. is your code using same algorithm?

    * It looks to be the same algorithm: compute `b` and `H`, solve the linear system `H dx = -b`, and update the poses.  What the included package contributes is a Graph SLAM solver that:

        * uses a standard data format (.g2o files) for importing and exporting data
        * has been validated against results from [g2o](https://github.com/RainerKuemmerle/g2o), which is arguably the de facto Graph SLAM solver

    * If the code in `graph_based_slam.py` could be used to achieve these same results, then I would agree that the code is redundant.  

2. Do you want to keep latex codes and pdf? is it impossible to use jupyter notebook to describe math?

    * Yes, I definitely think that both the LaTeX source and the PDF should be kept.  That way, it is easy to make a correction or contribution to the write-up.
    * I originally tried to put the LaTeX code in the Jupyter notebook, but the formatting was really bad and I couldn't make it work.  For example: https://github.com/JeffLIrion/PythonRobotics/blob/20fec7d60f9ca6063a58430be65c96f33beded28/SLAM/GraphBasedSLAM/graphSLAM_formulation.ipynb

3. You added images in the image dir. Are these impossible  to embedded a jupyter notebook?

    * I can embed them in a Jupyter notebook.  I also meant to make a .gif out of them.  I'll do that now.

**Comment by JeffLIrion** [comment]
For my own reference:

```
convert -background white -alpha remove -alpha off -delay 200 -loop 0 iteration*.png Graph_SLAM_optimization.gif
```

**Comment by AtsushiSakai** [comment]
> If the code in graph_based_slam.py could be used to achieve these same results, then I would agree that the code is redundant.
Do you think it is possible for you to update the graph_based_slam.py using your solver?

**Comment by AtsushiSakai** [comment]
>Yes, I definitely think that both the LaTeX source and the PDF should be kept. 

OK. I got it.

**Comment by AtsushiSakai** [comment]
> Let me know if you'd like me to delete the .png images.

Oh. great. Please remove the pngs. I want to keep the PR as simple as possible.

**Comment by JeffLIrion** [comment]
> > Let me know if you'd like me to delete the .png images.
> 
> 
> 
> Oh. great. Please remove the pngs. I want to keep the PR as simple as possible.

Done. I left only the .gif image.

**Comment by JeffLIrion** [comment]
> > If the code in graph_based_slam.py could be used to achieve these same results, then I would agree that the code is redundant.
> 
> 
> 
> Do you think it is possible for you to update the graph_based_slam.py using your solver?
> 
>  

I'm sure it's possible, but my wife and I just had a baby, so I simply don't have the time to do that right now. Sorry.

**Comment by AtsushiSakai** [comment]
OK. Thank you so much.
> I simply don't have the time to do that right now. Sorry.
No problem. Is it ok to merge this PR now? Maybe I can do it later.

**Comment by JeffLIrion** [comment]
> No problem. Is it ok to merge this PR now? Maybe I can do it later.

I'll squash my commits and then it will be ready.

**Comment by JeffLIrion** [comment]
I squashed the commits, so this is ready to be merged.

**Comment by AtsushiSakai** [comment]
@JeffLIrion Thank you so much for your great work. More PRs are always welcome!!

**Pull request merged**